taxiunn/driver_auth/models.py

```diff
@@ -0,0 +1,97 @@
+from django.db import models
+from django.contrib.auth.base_user import BaseUserManager, AbstractBaseUser
+from django.contrib.auth.backends import BaseBackend
+
+
+def validation(func):
+    """Функция-декоратор для валидации данных."""
+
+    def wrapper(self, email, password, *args, **kwargs):
+        """Валидация полей email и password."""
+        if email is None:
+            raise TypeError('Driver must have an email!')
+
+        if password is None:
+            raise TypeError('Driver must have a password!')
+
+        return func(self, email, password, *args, **kwargs)
+    return wrapper
+
+
+class DriverManager(BaseUserManager):
+    """Менеджер для пользователя класса Driver."""
+
+    @validation
+    def create_user(self, email, password):
```

Comment:
```suggestion
    def create_user(self, email: str, password: str) -> User:
```
тип пользователь в `django` не знаю как пишется написал `User`